lib/nerves_key/server.ex

```diff
@@ -0,0 +1,74 @@
+defmodule NervesKey.Server do
+  @moduledoc false
+  use GenServer
+
+  @doc false
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  @spec cache_device_cert(ATECC508A.Transport.t(), X509.Certificate.t()) :: :ok
+  def cache_device_cert(transport, cert) do
+    transport_info = ATECC508A.Transport.info(transport)
+
+    GenServer.cast(
+      __MODULE__,
+      {:cache_device_cert, {transport_info.bus_name, transport_info.address}, cert}
+    )
+
+    cert
```

Comment:
```suggestion
    transport_info = ATECC508A.Transport.info(transport)

    GenServer.cast(
      __MODULE__,
      {:cache_device_cert, {transport_info.bus_name, transport_info.address}, cert}
    )
```

Returning the cert doesn't match the spec. I'd say we just keep it to `:ok` and leave as best effort